src/internal/connector/onedrive/permission.go

```diff
@@ -113,58 +102,52 @@ func createRestoreFoldersWithPermissions(
 		service,
 		driveID,
 		id,
-		parentPermissions,
-		folderPermissions,
+		folderMetadata,
 		permissionIDMappings)
 
 	return id, err
 }
 
-// getChildPermissions is to filter out permissions present in the
-// parent from the ones that are available for child. This is
-// necessary as we store the nested permissions in the child. We
-// cannot avoid storing the nested permissions as it is possible that
-// a file in a folder can remove the nested permission that is present
-// on itself.
-func getChildPermissions(
-	childPermissions, parentPermissions []UserPermission,
+func getPermissionDiff(
+	after, before []UserPermission,
```

Comment:
My OCD is so confused.  Should before come before after?  Or should it maintain alphabetical order?

![my brain!](https://media.tenor.com/r9qSk38AiFcAAAAC/my-brain-hurts-mindblowing.gif)

Reply:
Haha, thanks for bringing this up. I had it in my todo list to change, but missed it. I too think should be `before` and then `after`.